File: solutions/typescript/secret-handshake/1/secret-handshake.ts
export function commands(number: number): string[] {
  const actions: string[] = [];
  
  // Check each bit position and add corresponding action
  if (number & 1) { // Bit 0: 00001
    actions.push("wink");
  }
  
  if (number & 2) { // Bit 1: 00010
    actions.push("double blink");
  }
  
  if (number & 4) { // Bit 2: 00100
    actions.push("close your eyes");
  }
  
  if (number & 8) { // Bit 3: 01000
    actions.push("jump");
  }
  
  if (number & 16) { // Bit 4: 10000 - Reverse the order
    actions.reverse();
  }
  
  return actions;
}

File: solutions/ruby/secret-handshake/1/secret_handshake.rb
class SecretHandshake
  # Mapeamos los bits con sus respectivas acciones
  ACTIONS = {
    1 => "wink",
    2 => "double blink",
    4 => "close your eyes",
    8 => "jump"
  }.freeze

  def initialize(number)
    @number = number
  end

  def commands
    # Validamos que sea un entero antes de procesar por bits
    return [] unless @number.is_a?(Integer)

    handshake = []

    # Iteramos por cada par bit-acción de nuestra constante
    ACTIONS.each do |bit, action|
      # Si el bit está activo en @number, agregamos la acción
      handshake << action if (@number & bit) != 0
    end

    # El bit 16 (10000 en binario) dictamina si debemos invertir el orden
    handshake.reverse! if (@number & 16) != 0

    handshake
  end
end